plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/quickfix/NumberConversionFix.kt

```diff
@@ -38,7 +44,22 @@ class NumberConversionFix(
 
     override fun invoke(project: Project, editor: Editor?, file: KtFile) {
         val element = element ?: return
-        val expressionToInsert = KtPsiFactory(file).createExpressionByPattern("$0.to$1()", element, typePresentation)
+        val psiFactory = KtPsiFactory(file)
+        val langVersion = element.languageVersionSettings.languageVersion
+        val expressionToInsert = when {
+            expressionTypeIsChar && langVersion >= LanguageVersion.KOTLIN_1_5 ->
+                if (expectedTypeIsInt) {
+                    psiFactory.createExpressionByPattern("$0.code", element)
+                } else {
+                    psiFactory.createExpressionByPattern("$0.code.to$1()", element, typePresentation)
+                }
+            expectedTypeIsChar && langVersion >= LanguageVersion.KOTLIN_1_5 ->
+                psiFactory.createExpressionByPattern("$0.toInt().toChar()", element)
```

Comment:
`expectedTypeIsChar && langVersion >= LanguageVersion.KOTLIN_1_5` should be changed to 
`!expectedTypeIsInt && expectedTypeIsChar && langVersion >= LanguageVersion.KOTLIN_1_5`

Reply:
otherwise the code will generate `anyInt.toInt().toChar()`